File: templates/problems/18_bit_manipulation/05_missing_number/solution.go
package missing_number

// solveMissingNumber uses XOR to find the missing number. O(n) time, O(1) space.
func solveMissingNumber(nums []int) int {
	result := len(nums)
	for i, v := range nums {
		result ^= i ^ v
	}
	return result
}

File: templates/problems/18_bit_manipulation/05_missing_number/typescript/solution.ts
/**
 * Use XOR to find the missing number. O(n) time, O(1) space.
 */
export function solveMissingNumber(nums: number[]): number {
    let result = nums.length;
    for (let i = 0; i < nums.length; i++) {
        result ^= i ^ nums[i];
    }
    return result;
}
